dbt_invoke/properties.py

```diff
@@ -206,6 +208,244 @@ def echo_macro(ctx):
+                definition['resource_index'] = next(
+                    (
+                        index
+                        for (index, x) in enumerate(
+                            current_property_yaml[
+                                definition['resource_type_plural']
+                            ]
+                        )
+                        if x['name'] == definition['resource_name']
+                    ),
+                    None,
+                )
+                definition['resource_properties'] = current_property_yaml[
+                    definition['resource_type_plural']
+                ][definition['resource_index']]
+                # Try to create each new property file. For each success, save
+                # the index of the resource to remove from the migration
+                # property file.
+                try:
+                    property_file_dict = _get_property_header(
+                        definition['resource_name'],
+                        definition['resource_type'],
+                        definition['resource_properties'],
+                    )
+                    _utils.write_yaml(
+                        definition['resource_destination'],
+                        property_file_dict,
+                        mode='x',
+                    )
+                    indices_to_remove[
+                        definition['resource_type_plural']
+                    ].append(definition['resource_index'])
+                    _LOGGER.info(
+                        f"Created {definition['resource_destination']}"
+                    )
+                except Exception as e:
+                    _LOGGER.error(
+                        f"Failed to create {definition['resource_destination']}: {e}"
+                    )
+        # Remove migrated resources from migration_property_file_dict
+        removed_counter = 0
+        for resource_type_plural, indices in indices_to_remove.items():
+            for i in sorted(indices, reverse=True):
+                current_property_yaml[resource_type_plural].pop(i)
+                removed_counter += 1
+        # Overwrite the migration property file with its remaining
+        # post-migration contents.
+        _utils.write_yaml(location, current_property_yaml)
+        _LOGGER.info(
+            f'Removed {str(removed_counter)} migrated resources from {str(location)}'
+        )
+        # Remove resource types from the source yaml
+        for resource_type_plural, resources in current_property_yaml.items():
+            if (
+                resource_type_plural in _SUPPORTED_RESOURCE_TYPES.values()
+                and len(resources) == 0
+            ):
+                current_property_yaml.pop(resource_type_plural)
+        # Check if there are any properties left in the source file.
+        # If there aren't then delete the file.
+        if Path(location).read_text().strip() == "":
```

Comment:
Just a note on this one. We never remove the ```version: 2``` from the header of the yaml.
Meaning it will never become empty.
This check should probable be 
```suggestion
        if Path(location).read_text().strip().lower() == "version: 2":
```

Reply:
Thanks for this! I ran into this too and actually made the same exact change locally as part of a handful of edits I plan to push today.